src/fs_core.ts

```diff
@@ -103,48 +104,65 @@ export class Directory {
   }
 
   get_entry_for_path(path: string): File | Directory | SyncOPFSFile | null {
-    let entry: File | Directory | SyncOPFSFile = this;
-    for (let component of path.split("/")) {
-      if (component == "") break;
-      if (component == ".") continue;
-      if (!(entry instanceof Directory)) {
-        return null;
-      }
-      if (entry.contents[component] != undefined) {
-        entry = entry.contents[component];
-      } else {
-        //console.log(component);
-        return null;
-      }
+    return getEntryForPath(path, this);
+  }
+
+  create_entry_for_path(
+    path: string,
+    is_dir: boolean,
+  ): File | Directory | SyncOPFSFile {
+    return createEntryForPath(path, is_dir, this);
+  }
+}
+
+function getEntryForPath(
+  path: string,
+  root: Directory,
+): File | Directory | SyncOPFSFile | null {
+  let entry: File | Directory | SyncOPFSFile = root;
+  for (const component of path.split("/")) {
+    if (component == "") break;
+    if (component == ".") continue;
+    if (!(entry instanceof Directory)) {
+      return null;
+    }
+    if (entry.contents[component] != undefined) {
+      entry = entry.contents[component];
+    } else {
+      //console.log(component);
+      return null;
     }
-    return entry;
-  }
-
-  create_entry_for_path(path: string, is_dir: boolean): File | Directory {
-    // FIXME fix type errors
-    let entry: File | Directory = this;
-    let components: Array<string> = path
-      .split("/")
-      .filter((component) => component != "/");
-    for (let i = 0; i < components.length; i++) {
-      let component = components[i];
-      // @ts-ignore
-      if (entry.contents[component] != undefined) {
-        // @ts-ignore
-        entry = entry.contents[component];
+  }
+  return entry;
+}
+
+function createEntryForPath(
```

Comment:
I don't think splitting this out a separate function improves anything.

Reply:
It (and similar extracted functions) avoids the eslint error about shadowing `this`.

Reply:
Huh, where is it shadowing `this`? Can you show the error reported by eslint?

Reply:
Sorry, not shadowing, aliasing: https://typescript-eslint.io/rules/no-this-alias/#examples

Reply:
I think I understand why this lint exists, but it doesn't apply to this case. I did vote for disabling this lint instead. The code movement makes it harder to review.